fix: kill entire process groups to prevent dev server zombie processes on timeout

sQVe opened the pull request:
### Summary

SST dev command was leaking child processes when timing out. Each timeout left an entire process tree running, eventually exhausting system resources. This fix ensures all spawned processes die with their parents.

#### Background
 
Our team continuously experiences memory exhaustion and database connection leaks from sst dev, even when idling. During normal development work (hot reloads, rebuilds), it spawns new processes without cleaning up old ones, accumulating 200+ processes consuming over 100% system memory in a single dev session.

#### The Bug

When `sst dev` times out (default 50 mins), it prints `[timeout]` but never kills the running process. The timeout handler in [mosaic.go:85-91](https://github.com/sst/sst/blob/9ced491d45eb971842c67e2b87ba43df817bc241/cmd/sst/mosaic.go#L85-L90) just continues the loop without calling `process.Kill()`. When the loop eventually restarts, only the parent dies - children survive as zombies. 🧟 

**Reproduction:**
```bash
#!/bin/bash
node -e "setInterval(() => console.log('child alive'), 2000)" &
wait
```

1. Lower the timeout in `mosaic.go` for debugging.
1. Run above script with `sst dev`
1. Wait for the timeout
1. Check `ps aux | grep node` for accumulating orphaned processes (parent PID is 1)

#### The Fix

1. **Create process groups:** Set `Setpgid: true` for spawned processes on Unix ([mosaic.go:126-130](https://github.com/sst/sst/pull/6133/files#diff-d596e671dba70e5c2adde994d2c251da1f11e48cc3b2179f0a97c1c00e6f4049R126-R130))
1. **Kill entire groups:** Use `syscall.Kill(-pid, signal)` to terminate all processes in group ([process.go:111](https://github.com/sst/sst/pull/6133/files#diff-d821ab8e2837e9704c2828e47ea4558eea087c46e1870d71f51842b5764b633cR111))
1. **Graceful fallback:** If group kill fails, fall back to individual process termination

#### Test Plan

- [x] Run `sst dev` with a script that spawns children, verify all processes die on timeout
- [x] Test Ctrl+C kills entire process tree
- [x] Verify `ps aux` shows no orphaned processes after SST exits
- [ ] Confirm Windows builds work (excluded from Unix-specific changes)
- [x] Check runtime processes (Python/Go/Node/Rust) still terminate correctly

sQVe committed:
fix: kill entire process groups to prevent dev server zombie processes on timeout

vimtor reviewed (changes requested):
the changes work on unix

before:

https://github.com/user-attachments/assets/6c12d717-386b-4d4d-bebf-13f525b9cf6f

after:

https://github.com/user-attachments/assets/32fea16c-ba17-4363-891d-cb9fe276cc78

but sadly the build breaks on windows

```sh
GOOS=windows GOARCH=amd64 go build ./pkg/process/

# github.com/sst/sst/v3/pkg/process
pkg/process/process.go:111:21: undefined: syscall.Kill
pkg/process/process.go:134:21: undefined: syscall.Kill
```

this happens because `syscall.Kill` is not available on windows

@sQVe could you extract the kill process logic into 2 separate files using go build directives

there are similar solutions in the repo, for example:
https://github.com/sst/sst/blob/dev/cmd/sst/mosaic/multiplexer/tcell-term/vt_unix.go
https://github.com/sst/sst/blob/dev/cmd/sst/mosaic/multiplexer/tcell-term/vt_windows.go

sQVe commented:
@vimtor Cheers for taking a look ❤️. I will look into that ASAP, either tonight or tomorrow! 👍🏼

sQVe committed:
refactor(process): split kill logic into platform-specific files

syscall.Kill doesn't exist on Windows, so move signal handling to
kill_unix.go and kill_windows.go. Also fixes a few bugs:
- guard against PID <= 0 (would signal our own process group)
- treat ESRCH as success (process already dead)
- continue to SIGKILL even if SIGTERM fails

sQVe committed:
refactor(process): don't clobber SysProcAttr in Detach

Detach() was overwriting any existing SysProcAttr. Now it only sets
Setpgid without nuking other fields. Also moved mosaic.go to use
the centralized Detach() helper.

sQVe committed:
fix(multiplexer): add Setpgid to PTY processes

PTY processes only had Setsid, so they escaped process group kills.
Adding Setpgid lets us terminate them properly on timeout.

sQVe committed:
chore(process): document Windows limitation and remove redundant nil check

sQVe committed:
fix(process): only log 'untracked process' when process wasn't found

Previously the log message fired unconditionally after the cleanup loop,
even when the process was successfully removed from the tracked list.

sQVe committed:
fix(process): ensure cleanup runs on all error paths

Previously, error returns from sendKillSignal or timeout would skip
cleanup, leaving the process in the tracked cmds slice (memory leak).
Now all paths go through cleanup before returning.

vimtor committed:
refactor kill logic, fix ESRCH bug, add tests

- remove goto from Kill(), extract escalatingKill/forceKill/untrack
- fix sendSignal treating ESRCH as "process dead" when it means
  "no process group" — now falls back to individual signal
- drop redundant Setpgid from vt_unix.go (Setsid already creates group)
- add process package tests

vimtor committed:
add newline before timeout message

vimtor committed:
fix test for macOS and clarify untrack log message

vimtor reviewed (approved):
thanks for your contribution @sQVe 

made some minor changes and added tests

<details>

<summary>proof of it working</summary>

<img width="1258" height="1310" alt="Image" src="https://github.com/user-attachments/assets/7cf27e9d-adf1-4418-91d0-76485c5a6a0f" />

<img width="1608" height="452" alt="Image" src="https://github.com/user-attachments/assets/8802ba1b-794d-4e6c-b34a-9f908130e316" />

</details>
